# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/builtin/RegExp.h
# Commit: a468786b4570
# Full Hash: a468786b4570997bb28c6ed03acf84f2ffef71df
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2020-05-22 21:38:53
# Regressor Bug: 1362154
# File Overlap Count: 1
# Description:
#   Bug 1362154: Part 8: Named capture replacements in RegExpGetSubstitution r=arai
#   
#   This patch implements the changes to GetSubstitution to support named captures.
#   
#   See https://tc39.es/proposal-regexp-named-groups/#sec-getsubstitution
# ==============================================================================

diff -r c296b98a4e22 -r a468786b4570 js/src/builtin/RegExp.h
--- a/js/src/builtin/RegExp.h	Thu May 21 04:52:12 2020 +0000
+++ b/js/src/builtin/RegExp.h	Thu May 21 16:57:07 2020 +0000
@@ -107,7 +107,7 @@
 extern MOZ_MUST_USE bool RegExpGetSubstitution(
     JSContext* cx, HandleArrayObject matchResult, HandleLinearString string,
     size_t position, HandleLinearString replacement, size_t firstDollarIndex,
-    MutableHandleValue rval);
+    HandleValue namedCaptures, MutableHandleValue rval);
 
 extern MOZ_MUST_USE bool GetFirstDollarIndex(JSContext* cx, unsigned argc,
                                              Value* vp);